core/mdns_browser.go

```diff
@@ -0,0 +1,56 @@
+package core
+
+import (
+	"errors"
+	"log"
+	"time"
+
+	"github.com/godbus/dbus/v5"
+	"github.com/holoplot/go-avahi"
+)
+
+// Search the for the service with the specified instance name
+func searchService(sb *avahi.ServiceBrowser, srv *avahi.Server, name string, t int) (avahi.Service, error) {
+	var s avahi.Service
+	for {
+		select {
+		case s = <-sb.AddChannel:
+			s, err := srv.ResolveService(s.Interface, s.Protocol, s.Name,
+				s.Type, s.Domain, avahi.ProtoUnspec, 0)
+			if err != nil {
+				log.Fatalf("ResolveService() failed: %v", err)
```

Comment:
You are in a package. Never exit from a function in a package. Leave this up to the package user whether to treat this as a fatal error!